### PR TITLE
Feature/thorough payload checking

### PR DESCRIPTION
**Pull request by DanielReinagel**
## Description

There were many cases in which we were not giving good error messages when given a bad payload. There were a couple of instances in which we responded with 500 when a bad payload was sent to the backend. There were a couple of instances in which we responded with 400 even though the issue might be a server error. I fixed as many of these issues in regards to processing the request payload as I could find.

Fixes # (issue)

https://trello.com/c/uQaI2QEs/143-when-checking-payloads-the-be-should-check-for-every-way-that-the-payload-can-be-inputted-incorrectly-before-sending-it-to-the-d

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [ ] No duplicate code left within changed files
- [ ] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

## Details:

video coming soon

adminMiddleware properly checking for a bad payload and responding with a useful error
adminRouter return 500 instead of 400 when the server has failed to create an admin
adminRouter renamed updaedAdmin to updatedAdmin
adminRouter return 500 instead of 400 when the server has failed to update an admin

api/functions/formatChecking created the file and made two functions that are exported. One checks to see if a time string is properly formatted. The other checks if a date string is properly formatted. Both return a boolean.

classInstancesMiddleware checkClassInstanceObject properly checking for a bad payload and responding with a userful error in each case of a bad request.

courseTypesMiddleware checkCoursePayload properly checking for a bad payload and responding with a userful error in each case of a bad request.

instructorMiddleware created checkInstructorObject that properly checking for a bad payload and responding with a userful error in each case of a bad request.
instructorRouter hooked up checkInstructorObject

added parentMiddleware.js with checkParentObject that functions
parentRouter hooked up with checkParentObject

added profileMiddleware.js with checkProfileObject that functions
profileRouter hooked up with checkProfileObject
profileRouter minorly editted to clean code that is no longer necessary due to checkProfileObject

### CODE REVIEW

**Review by jonahjtr — APPROVED** [review]
great work on simplifying and consolidating if statements

**Review by LeonelHays — APPROVED** [review]
good job man! That is a lot of middleware and all of it seems correct and clean

File: api/admin/adminRouter.js [hunk 1]
```diff
@@ -27,19 +27,19 @@ router.post('/', checkPayload, authRequired, async (req, res, next) => {
     const newadmin = await Admins.addAdmin(req.body);
     res.status(201).json(newadmin);
   } catch (error) {
-    next({ status: 400, message: 'could not create new admin profile' });
+    next({ status: 500, message: 'could not create new admin profile' });
   }
```
Comment (JesusNavarrete215) [inline api/admin/adminRouter.js:31]: Nice change from status 400 to 500 to show that the error is from the server and not the user.

**Review by JesusNavarrete215 — COMMENTED** [review]
Overall the code looks clean and minor changes have been done to take consideration of small details.

**Pull request merged**